File: easy-solution/Antidote21/414.java
class Solution {
    public int thirdMax(int[] nums) {
        Arrays.sort(nums);
        int prev = nums[0];
        int index = 1;
        for (int i=1; i<nums.length; i++) {
            if (prev == nums[i]) {
                continue;
            }
            prev = nums[i];
            nums[index++] = nums[i]; 
        }

        if (index < 3) {
            return nums[index-1];
        }
        
        return nums[index-3];
    }
}

File: easy-solution/minesp3164/python/414.py
class Solution(object):
    def thirdMax(self, nums):
        setnums = sorted(set(nums))
        if len(setnums) >=3:
            return setnums[len(setnums) - 3]
        else:
            return setnums[-1]
"""
class Solution(object):
    def thirdMax(self, nums):
        nums.sort()
        setnums = list(set(nums))
        if len(setnums) >=3:
            return setnums[len(setnums) - 3]
        else:
            return setnums[-1]

"""


